android/app/src/main/java/org/electroncash/electroncash3/ColdLoad.kt

```diff
@@ -0,0 +1,105 @@
+
+    //    val coldloadDialog by lazy { targetFragment as ColdLoadDialog }
+    val model: Model by viewModels()
+
+
+    init {
+        if (daemonModel.wallet!!.callAttr("is_watching_only").toBoolean()) {
+            throw ToastException(R.string.this_wallet_is)
+        } else if (daemonModel.wallet!!.callAttr("get_receiving_addresses")
+                        .asList().isEmpty()) {
+            // At least one receiving address is needed to call wallet.dummy_address.
+            throw ToastException(
+                    R.string.electron_cash_is_generating_your_addresses__please_wait_)
+        }
+    }
+
+    override fun onBuildDialog(builder: AlertDialog.Builder) {
+        builder.setTitle(R.string.load_unbroadcasted)
+                .setView(R.layout.load)
+                .setNegativeButton(android.R.string.cancel, null)
+                .setNeutralButton(R.string.qr_code, null)
+                .setPositiveButton(android.R.string.ok, null)
+    }
+
+    override fun onShowDialog() {
+        super.onShowDialog()
+        val ourClipboard = getSystemService(ClipboardManager::class)
+
+        etTransaction.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
+            override fun afterTextChanged(s: Editable?) {
+                val currenttext = etTransaction.text
+                //checks if text is blank. further validations can be added here
+                dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = when { currenttext.isNotBlank() -> {true} else -> false }
+            }
+
+        })
+        dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener { onOK() }
+        dialog.getButton(AlertDialog.BUTTON_NEUTRAL).setOnClickListener { scanQR(this) }
+        dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = false
+        btnPaste.isEnabled = when { !ourClipboard.hasPrimaryClip() -> { false } else -> {true}}
+        btnPaste.setOnClickListener {
+            val clipdata = ourClipboard.primaryClip
+            val cliptext = clipdata!!.getItemAt(0)
+            etTransaction.setText(cliptext.text)
+
+        }
+    }
+
+
+    // Receives the result of a QR scan.
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        val result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data)
+        if (result != null && result.contents != null) {
+            onScan(result.contents)
+        } else {
+            super.onActivityResult(requestCode, resultCode, data)
+        }
+    }
+
+    fun onScan(uri: String) {
+
+        etTransaction.setText(uri)
+    }
+
+    fun onOK() {
+        // try to send user input to network to be broadcast,
+        // this should work even if tx is not vaild transaction, but nothing happens
```

Comment:
If there's an error, then we should display it rather than silently failing. It looks like `broadcast_transaction` reports failure by returning (False, error_msg), while `broadcast_transaction2` throws an exception, which might be more convenient.

Reply:
I changed it to broadcast_transaction2, but it crashes (while showing the error I wanted toasted) instead of just toasting it, so I changed it back for now. Will try to figure out how to use that such that it doesn't crash.

Reply:
also with broadcast_transaction2, when I pass it a valid transaction, it still crashes, but at least not before it successfully broadcasts it. I'll figure out what I'm doing wrong

Reply:
OK, let me know when you've fixed it and I'll test it myself. Note that no Python function will directly throw `ToastException`, they'll throw `PyException`, and there are a couple of examples of this in ColdSend.kt.

The crash stack trace should be visible in the Android Studio Logcat window. If you have trouble understanding it then just post it here and I'll have a look.

Reply:
Here is the error I get when I use broadcast_transaction2 Although it crashes, the transaction does end up getting broadcast. 

`2020-09-18 12:29:26.045 7958-7958/org.electroncash.wallet E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.electroncash.wallet, PID: 7958
    com.chaquo.python.PyException: AttributeError: 'str' object has no attribute 'txid'
        at <python>.electroncash.network.broadcast_transaction2(network.py:1824)
        at <python>.chaquopy_java.call(chaquopy_java.pyx:283)
        at <python>.chaquopy_java.Java_com_chaquo_python_PyObject_callAttrThrows(chaquopy_java.pyx:255)
        at com.chaquo.python.PyObject.callAttrThrows(Native Method)
        at com.chaquo.python.PyObject.callAttr(PyObject.java:207)
        at org.electroncash.electroncash3.ColdLoadDialog.onOK(ColdLoad.kt:75)
        at org.electroncash.electroncash3.ColdLoadDialog$onShowDialog$2.onClick(ColdLoad.kt:49)
        at android.view.View.performClick(View.java:6256)
        at android.view.View$PerformClick.run(View.java:24701)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)`

Reply:
Here is the error when I input a something that isn't a transaction
`2020-09-18 12:37:41.513 8233-8233/org.electroncash.wallet E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.electroncash.wallet, PID: 8233
    com.chaquo.python.PyException: ServerErrorResponse: An error occurred broadcasting the transaction
        at <python>.electroncash.network.broadcast_transaction2(network.py:1822)
        at <python>.chaquopy_java.call(chaquopy_java.pyx:283)
        at <python>.chaquopy_java.Java_com_chaquo_python_PyObject_callAttrThrows(chaquopy_java.pyx:255)
        at com.chaquo.python.PyObject.callAttrThrows(Native Method)
        at com.chaquo.python.PyObject.callAttr(PyObject.java:207)
        at org.electroncash.electroncash3.ColdLoadDialog.onOK(ColdLoad.kt:75)
        at org.electroncash.electroncash3.ColdLoadDialog$onShowDialog$2.onClick(ColdLoad.kt:49)
        at android.view.View.performClick(View.java:6256)
        at android.view.View$PerformClick.run(View.java:24701)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
2020-09-18 12:37:41.513 8233-8233/org.electroncash.wallet E/ACRA: ACRA caught a PyException for org.electroncash.wallet
    com.chaquo.python.PyException: ServerErrorResponse: An error occurred broadcasting the transaction
        at <python>.electroncash.network.broadcast_transaction2(network.py:1822)
        at <python>.chaquopy_java.call(chaquopy_java.pyx:283)
        at <python>.chaquopy_java.Java_com_chaquo_python_PyObject_callAttrThrows(chaquopy_java.pyx:255)
        at com.chaquo.python.PyObject.callAttrThrows(Native Method)
        at com.chaquo.python.PyObject.callAttr(PyObject.java:207)
        at org.electroncash.electroncash3.ColdLoadDialog.onOK(ColdLoad.kt:75)
        at org.electroncash.electroncash3.ColdLoadDialog$onShowDialog$2.onClick(ColdLoad.kt:49)
        at android.view.View.performClick(View.java:6256)
        at android.view.View$PerformClick.run(View.java:24701)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)`

Reply:
I changed it so it toasts any errors, and toasts success if it works.